src/ghostfragment/fragmenting/bond_based.cpp

```diff
@@ -26,6 +26,23 @@ using subset_type = typename result_type::nucleus_index_set;
 using input_type  = pt::NuclearGraphToFragmentsTraits::graph_type;
 using nuclei_type = typename input_type::nuclei_type;
 
+std::ostream& operator<<(std::ostream& os, const std::vector<long unsigned int, std::allocator<long unsigned int> >& value) {
+    os << value << std::endl;
+    return os;
+}
+
+std::ostream& operator<<(std::ostream& os, const std::vector<subset_type>& mol) {
+    for (subset_type value : mol) {
+        os << value << std::endl;
+    }
+    return os;
+}
+
+std::ostream& operator<<(std::ostream& os, const chemist::fragmenting::FragmentedNuclei<chemist::Nuclei>& mol) {
+    os << mol << std::endl;
```

Comment:
Your segfault is caused by infinite recursion. This line calls the function you are implementing from within the implementation, i.e., you are trying to implement a print function by calling the print function you are implementing.